shell/platform/darwin/macos/framework/Source/FlutterViewController.mm

```diff
@@ -526,14 +536,26 @@ - (NSDictionary*)getClipboardData:(NSString*)format {
 }
 
 - (void)setClipboardData:(NSDictionary*)data {
-  NSPasteboard* pasteboard = [NSPasteboard generalPasteboard];
+  NSPasteboard* pasteboard = [self getPasteboard];
   NSString* text = data[@"text"];
+  [pasteboard clearContents];
   if (text && ![text isEqual:[NSNull null]]) {
-    [pasteboard clearContents];
     [pasteboard setString:text forType:NSPasteboardTypeString];
   }
 }
 
+- (NSDictionary*)clipboardHasStrings {
```

Comment:
nit: It's a bit odd having a method called `clipboardHasStrings` return something other than a bool.  It is probably better to return bool and wrap it into a dictionary at the call site, imo.

Reply:
That's a good call, the docs I wrote made this even more confusing actually.